src/matrix/matrixnormal.jl

```diff
@@ -0,0 +1,159 @@
+"""
+    MatrixNormal{T <: Real, ST <: AbstractPDMat} <: ContinuousMatrixDistribution
+    
+Matrix Normal Distribution
+
+`X ~ MN(M, U, V)`
+
+M: n x p
+
+U: n x n positive definite
+
+V: p x p positive definite
+
+f(X) = c0 * exp( -0.5 tr[inv(V) (X - M)' inv(U) (X - M)] )
+
+c0   = (2pi) ^ {-np / 2} |V| ^ {-n / 2} |U| ^ {-p / 2}
+
+https://en.wikipedia.org/wiki/Matrix_normal_distribution
+
+"""
+struct MatrixNormal{T <: Real, ST <: AbstractPDMat} <: ContinuousMatrixDistribution
```

Comment:
Do the types of `U` and `V` really have to be exactly the same subtype of `AbstractPDMat`?

Reply:
No, but in every instance where they're used, they're used together, and you will get promotion to the common element type anyway:
```julia
#  logpdf
-0.5 * tr( (d.V \ At) * (d.U \ A) ) 

#  rand
A .= d.M + d.U.chol.L * X * d.V.chol.U
```